[BUG] pip install bug

FastGeodis/geodis_fastmarch.cpp: In functiona'at::Tensor gradient2d(at::Tensor)':
FastGeodis/geodis_fastmarch.cpp:87:46: error: 'kSame' is not a member of 'torch
87
.padding(torch::kSame)
FastGeodis/geodis_fastmarch.cpp:95:46: error: 'kSame' is not a member of 'torch'
95
.padding(torch::kSame)
FastGeodis/geodis_fastmarch.cpp: In function 'at::Tensor gradient3d(at::Tensor)':
FastGeodis/geodis_fastmarch.cpp:281:46: error: 'kSame' is not a member of 'torch'
281
.padding(torch::kSame)
FastGeodis/geodis_fastmarch.cpp:289:46: error: 'kSame' is not a member of
N'torch
289
.padding(torch::kSame)
FastGeodis/geodis_fastmarch.cpp:297:46: error: 'kSame' is not a member of 'torch'
297
.padding(torch::kSame)


![623b5dd5-666d-4b3a-b202-74bba324e54e](https://github.com/masadcv/FastGeodis/assets/8953923/9151edb3-c6ef-4120-822a-6e81f3d457ed)
![00dc13a5-9cab-4833-acc1-2abbd8eec94d](https://github.com/masadcv/FastGeodis/assets/8953923/e04a5755-b896-4ca3-a403-fa7155d0ae13)

--- a/FastGeodis/geodis_fastmarch.cpp
+++ b/FastGeodis/geodis_fastmarch.cpp
@@ -80,22 +80,26 @@ torch::Tensor gradient2d(torch::Tensor image)
     gx = torch::tile(gx, {channel, 1, 1, 1});
     gy = torch::tile(gy, {channel, 1, 1, 1});
 
-    auto imagegx = F::conv2d(image, 
+    // padding to enable 'same' padding for even kernel size
+    // help from: https://github.com/masadcv/PyTorchSamePaddingExplainer
+    auto imagepadx = torch::constant_pad_nd(image, {0, 1, 0, 0}, 0);
+    auto imagepady = torch::constant_pad_nd(image, {0, 0, 0, 1}, 0);
+
+    // dx and dy gradients
+    auto imagegx = F::conv2d(imagepadx, 
                              gx, 
                              F::Conv2dFuncOptions()
                              .stride(1)
-                             .padding(torch::kSame)
                              .groups(channel)
                             );
-
-    auto imagegy = F::conv2d(image, 
+    auto imagegy = F::conv2d(imagepady, 
                              gy, 
                              F::Conv2dFuncOptions()
                              .stride(1)
-                             .padding(torch::kSame)
                              .groups(channel)
                             );
     
+    // combine dx and dy to get gradient
     auto imagegrad = 0.5 * (
         torch::abs(imagegx) + 
         torch::abs(imagegy)
@@ -274,30 +278,35 @@ torch::Tensor gradient3d(torch::Tensor image)
     gy = torch::tile(gy, {channel, 1, 1, 1, 1});
     gz = torch::tile(gz, {channel, 1, 1, 1, 1});
 
-    auto imagegx = F::conv3d(image, 
+    // padding to enable 'same' padding for even kernel size
+    // help from: https://github.com/masadcv/PyTorchSamePaddingExplainer
+    auto imagepadx = torch::constant_pad_nd(image, {0, 1, 0, 0, 0, 0}, 0);
+    auto imagepady = torch::constant_pad_nd(image, {0, 0, 0, 1, 0, 0}, 0);
+    auto imagepadz = torch::constant_pad_nd(image, {0, 0, 0, 0, 0, 1}, 0);
+
+    // dx, dy and dz gradients
+    auto imagegx = F::conv3d(imagepadx, 
                              gx, 
                              F::Conv3dFuncOptions()
                              .stride(1)
-                             .padding(torch::kSame)
                              .groups(channel)
                             );
 
-    auto imagegy = F::conv3d(image, 
+    auto imagegy = F::conv3d(imagepady, 
                              gy, 
                              F::Conv3dFuncOptions()
                              .stride(1)
-                             .padding(torch::kSame)
                              .groups(channel)
                             );
 
-    auto imagegz = F::conv3d(image, 
+    auto imagegz = F::conv3d(imagepadz, 
                              gz, 
                              F::Conv3dFuncOptions()
                              .stride(1)
-                             .padding(torch::kSame)
                              .groups(channel)
                             );
     
+    // combine dx, dy and dz to get gradient
     auto imagegrad = (1.0/3.0) * (
         torch::abs(imagegx) + 
         torch::abs(imagegy) + 

--- a/setup.py
+++ b/setup.py
@@ -125,7 +125,7 @@ dep_dir = os.path.join(".", "dependency")
 
 setup(
     name="FastGeodis",
-    version="1.0.3",
+    version="1.0.4",
     description="Fast Implementation of Generalised Geodesic Distance Transform for CPU (OpenMP) and GPU (CUDA)",
     long_description=long_description,
     long_description_content_type="text/markdown",

